### PR TITLE
Fix addmm test & FusingOptimizer

### PR DESCRIPTION
**Pull request by zrphercule**
We did not have fusing of aten::mm in our previous version, which makes addmm test doesnt work in our newest version of pytorch. here we add it in and fixed the operator creating process.

### CODE REVIEW

File: torch_glow/src/FusingOptimizer.cpp [hunk 1]
```diff
@@ -35,6 +35,12 @@ graph(%input, %weight, %bias, %4):
   %output = aten::matmul(%input, %weight_t)
   %res = aten::add_(%output, %bias, %4)
   return (%res))IR";
+  std::string mm_add_pattern = R"IR(
+graph(%input, %weight, %bias, %4):
```
Comment (yinghai) [inline torch_glow/src/FusingOptimizer.cpp]: weird indent?

Comment (zrphercule) [reply torch_glow/src/FusingOptimizer.cpp]: Actually not, it is hardcoded IR pattern, and can only use this indent.

**Comment by jackm321** [comment]
Thanks for fixing it!
I'm confused why PyTorch CI passed, there is an assert in jitVsGlow that requires the expected nodes to be lowered to be specified.

File: torch_glow/src/PyTorchModelLoader.cpp [hunk 2]
```diff
@@ -684,23 +684,29 @@ llvm::Error PyTorchModelLoader::loadLinear(const torch::jit::Node *ptNode) {
...
+  auto *mul = F_.createMatMul("linear_mul", input, weights_t->getResult());
 
   if (hasGlowNodeValue(inputs[LinearInputs::bias])) {
     glow::NodeValue bias;
     ASSIGN_VALUE_OR_RETURN_ERR(bias,
                                getGlowNodeValue(inputs[LinearInputs::bias]));
-    return addGlowNodeValue(outputs[0], F_.createAdd("linear_add", mul, bias));
+    return addGlowNodeValue(
+        outputs[0],
+        (F_.createAdd("linear_add", mul->getResult(), bias))->getResult());
```
Comment (jackm321) [inline torch_glow/src/PyTorchModelLoader.cpp]: might be cleaneer to just put the createAdd above

File: torch_glow/src/PyTorchModelLoader.cpp [hunk 3]
```diff
@@ -684,23 +684,29 @@ llvm::Error PyTorchModelLoader::loadLinear(const torch::jit::Node *ptNode) {
...
   glow::NodeValue weights;
   ASSIGN_VALUE_OR_RETURN_ERR(weights,
                              getGlowNodeValue(inputs[LinearInputs::weights]));
 
-  auto *mul = F_.createMatMul("linear_mul", input, weights);
+  // Currently in Pytorch => Glow, we only translate t()+addmm(), t()+mm() and
+  // t()+matmul() to linear, Which means we need a transpose anyway, and also
+  // since addmm and mm only works for 2-D, we dont need to consider other
+  // transpose implementation.
+  auto *weights_t = F_.createTranspose("weights_t", weights, {1, 0});
```
Comment (jackm321) [inline torch_glow/src/PyTorchModelLoader.cpp]: can just put this in weights

File: torch_glow/src/PyTorchModelLoader.cpp [hunk 4]
```diff
@@ -684,23 +684,29 @@ llvm::Error PyTorchModelLoader::loadLinear(const torch::jit::Node *ptNode) {
...
-
   glow::NodeValue input;
   ASSIGN_VALUE_OR_RETURN_ERR(input,
                              getGlowNodeValue(inputs[LinearInputs::input]));
   glow::NodeValue weights;
   ASSIGN_VALUE_OR_RETURN_ERR(weights,
                              getGlowNodeValue(inputs[LinearInputs::weights]));
 
-  auto *mul = F_.createMatMul("linear_mul", input, weights);
+  // Currently in Pytorch => Glow, we only translate t()+addmm(), t()+mm() and
```
Comment (jackm321) [inline torch_glow/src/PyTorchModelLoader.cpp]: This is some pretty baked-in logic, we'll have to revisit this later probably.

Comment (zrphercule) [reply torch_glow/src/PyTorchModelLoader.cpp]: Exactly, that's why I left some comments here...

File: torch_glow/src/PyTorchModelLoader.cpp (hunk 2 above)
Comment (zrphercule) [reply torch_glow/src/PyTorchModelLoader.cpp]: ok

File: torch_glow/src/PyTorchModelLoader.cpp (hunk 3 above)
Comment (zrphercule) [reply torch_glow/src/PyTorchModelLoader.cpp]: what do you mean?

Comment (jackm321) [reply torch_glow/src/PyTorchModelLoader.cpp]: no need to define a new weights_t

**Comment by jackm321** [comment]
Please make sure to update jitVsGlow in addmm_test.py before landing

File: torch_glow/src/PyTorchModelLoader.cpp (hunk 3 above)
Comment (zrphercule) [reply torch_glow/src/PyTorchModelLoader.cpp]: oh... I see, anyway no big deal hahaha

**Comment by zrphercule** [comment]
> Please make sure to update jitVsGlow in addmm_test.py before landing

The reason why jitVsGlow passed at first is because when we first added the test, addmm is still "aten::addmm", but now somehow it is "aten::mm + aten::add" which caused this bug. Also you remember you fixed me a bug which makes this test doesnt really run the addmm op right? And that is why when we update pytorch version here, we didnt find out it was wrong...

Guess there is no need to change jitVsGlow?...

**Pull request closed without merging**